contact.cpp

```diff
@@ -0,0 +1,107 @@
+/*ID: ohs.han1
+TASK: contact
+LANG: C++
+ */
+
+#include<iostream>
+#include<cstdio>
+#include<cstring>
+#include<map>
+#include<algorithm>
+#include<fstream>
+using namespace std;
+
+const int MAXN = 400000;
+
+ifstream fin("contact.in");
+ofstream fout("contact.out");
+
+typedef struct
+{
+  string pattern;
+  int count;
+}frequencies;
+frequencies ar[MAXN];
```

Comment:
maybe you should've tried a vector, or an array of vectors?

you could make it so that ar[7] was a vector that contained the patterns that had frequency 7.